### ISSUE TITLE
More GD(Gradient descent) methode

### ISSUE THREAD

**Issue by LuposX** [LuposX/BostonHousingPrediction#31]
Implement more GD methods and compare them and let the user choose which he want to use:
*   GD(over whole dataset)
*   SGD(with batch size. Randomize after every epoch!)
*   GD with momentum
*   GD with adaptive learning rate
